sonar-plugin-api/src/main/java/org/sonar/api/batch/fs/internal/TestInputFileBuilder.java

```diff
@@ -52,24 +68,54 @@
   private int nonBlankLines;
   private int[] originalLineOffsets;
   private boolean publish = true;
+  private String contents;
 
+  /**
+   * Create a InputFile identified by the given module key and relative path.
+   * The module key will also be used as the module's base directory. 
+   */
   public TestInputFileBuilder(String moduleKey, String relativePath) {
     this(moduleKey, relativePath, batchId++);
   }
 
+  /**
+   * Create a InputFile with a given module key and module base directory. 
+   * The relative path is generated comparing the file path to the module base directory. 
+   * filePath must point to a file that is within the module base directory.
+   */
+  public TestInputFileBuilder(String moduleKey, File moduleBaseDir, File filePath) {
+    String relativePath = moduleBaseDir.toPath().relativize(filePath.toPath()).toString();
+    this.moduleKey = moduleKey;
+    setModuleBaseDir(moduleBaseDir.toPath());
+    this.relativePath = PathUtils.sanitize(relativePath);
+    this.id = batchId++;
+  }
+
   public TestInputFileBuilder(String moduleKey, String relativePath, int id) {
     this.moduleKey = moduleKey;
-    this.moduleBaseDir = Paths.get(moduleKey);
+    setModuleBaseDir(Paths.get(moduleKey));
     this.relativePath = PathUtils.sanitize(relativePath);
     this.id = id;
   }
 
+  public static TestInputFileBuilder create(String moduleKey, File moduleBaseDir, File filePath) {
+    return new TestInputFileBuilder(moduleKey, moduleBaseDir, filePath);
+  }
+
+  public static TestInputFileBuilder create(String moduleKey, String relativePath) {
+    return new TestInputFileBuilder(moduleKey, relativePath);
+  }
+
   public static int nextBatchId() {
     return batchId++;
   }
 
   public TestInputFileBuilder setModuleBaseDir(Path moduleBaseDir) {
-    this.moduleBaseDir = moduleBaseDir.normalize();
+    try {
+      this.moduleBaseDir = moduleBaseDir.normalize().toRealPath(new LinkOption[] {LinkOption.NOFOLLOW_LINKS});
+    } catch (IOException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)